### PR TITLE
Implement route support for http request metrics (#34)

### LINKED ISSUE #34: The value of path label should stay in pattern defination
Assuming we have such rest-style route defination:
```kotlin 
this.post("/api/v1/user/:userId").handler{ it.response().end("ok") }
this.post("/api/v1/timeline/:timelineId/comment/:commentId").handler{ it.response().end("ok") }
```

Currently when there's a request with new id hitting the server, there will be at least one new item generated.

```
vertx_http_server_requests{local="0.0.0.0:8080",path="/api/v1/user/1",remote="_",} 0.0
vertx_http_server_requests{local="0.0.0.0:8080",path="/api/v1/user/2",remote="_",} 0.0
vertx_http_server_requests{local="0.0.0.0:8080",path="/api/v1/user/3",remote="_",} 0.0
vertx_http_server_requests{local="0.0.0.0:8080",path="/api/v1/timeline/a01/comment/1",remote="_",} 0.0
vertx_http_server_requests{local="0.0.0.0:8080",path="/api/v1/timeline/a01/comment/2",remote="_",} 0.0
vertx_http_server_requests{local="0.0.0.0:8080",path="/api/v1/timeline/a02/comment/1",remote="_",} 0.0
vertx_http_server_requests{local="0.0.0.0:8080",path="/api/v1/timeline/a02/comment/3",remote="_",} 0.0
vertx_http_server_requests{local="0.0.0.0:8080",path="/api/v1/timeline/a02/comment/4",remote="_",} 0.0
```

I hope it can be like this:
```
vertx_http_server_requests{local="0.0.0.0:8080",path="/api/v1/registry/user/:userId",remote="_",} 0.0
vertx_http_server_requests{local="0.0.0.0:8080",path="/api/v1/timeline/:timelineId/comment/:commentId",remote="_",} 0.0
```

### PR DESCRIPTION
**Pull request by jekkel**
fixes #34 

Companion PRs:
* required: https://github.com/eclipse-vertx/vert.x/pull/3532
* https://github.com/vert-x3/vertx-web/pull/1682
* https://github.com/vert-x3/vertx-dropwizard-metrics/pull/101

### CODE REVIEW

**Comment by jekkel** [comment]
@jotak would you mind having a look? Both vert.x core and vert.x web PRs are merged already.

**Comment by jotak** [comment]
Hi @jekkel , sorry I was on a break, I will take a look asap

File: src/main/java/io/vertx/micrometer/impl/VertxHttpServerMetrics.java [hunk 1]
```diff
@@ -28,7 +28,11 @@
 import io.vertx.micrometer.impl.meters.Gauges;
 import io.vertx.micrometer.impl.meters.Timers;
 
+import java.util.ArrayList;
```
Comment (jotak) [inline src/main/java/io/vertx/micrometer/impl/VertxHttpServerMetrics.java:31]: Some of these imports are unused

**Comment by jotak** [comment]
Some tests are failing in `VertxHttpClientServerMetricsTest` , function `shouldReportHttpServerMetrics`. Some of the failures are related to your changes, and some others weren't ; I've fixed the latter in master so you should be able to fix the former after rebasing from master.

Other than that, the PR looks good to me!

File: src/main/java/io/vertx/micrometer/impl/VertxHttpServerMetrics.java (hunk 1 above)
Comment (jekkel) [reply src/main/java/io/vertx/micrometer/impl/VertxHttpServerMetrics.java:31]: Good find! Cleaning up.

**Comment by jekkel** [comment]
Thanks for the review! I rebased from master, fixed the test and the documentation. There has been a wrong metric name in the docs: `vertx_http_server_processingTime` but the source code uses `vertx_http_server_responseTime` so I fixed that when adding the new `route` label to the docs.

**Comment by jekkel** [comment]
I wonder whether this also resolved #86 ?

**Review by jotak — APPROVED** [review]
LGTM, thank you @jekkel !

**Comment by jotak** [comment]
> I wonder whether this also resolves #86 ?

Yes there was several issues discussed in #86 and indeed there was the problem of solving path label cardinality issue; with one solution being to configure some regex-based filtering (as described here https://github.com/vert-x3/vertx-micrometer-metrics/issues/86#issuecomment-556003656 ) ; and this PR now also provides a good alternative.

**Pull request merged**